terminus/geometry/latlon.py

```diff
@@ -153,3 +170,83 @@ def is_inside(self, origin, corner):
             if (self.lat - origin.lat) * (self.lat - corner.lat) <= 0 and \
                (self.lon - origin.lon) * (self.lon - corner.lon) >= 0:
                 return True
+
+    def _latitude_in_radians(self):
+        return math.radians(self.lat)
+
+    def _longitude_in_radians(self):
+        return math.radians(self.lon)
+
+    def _sin_lat(self):
+        return math.sin(self._latitude_in_radians())
+
+    def _sin_lon(self):
+        return math.sin(self._longitude_in_radians())
+
+    def _cos_lat(self):
+        return math.cos(self._latitude_in_radians())
+
+    def _cos_lon(self):
+        return math.cos(self._longitude_in_radians())
+
+
+class Ecef(object):
+
+    def __init__(self, x, y, z):
+        self.x = x
+        self.y = y
+        self.z = z
+
+    @classmethod
+    def equatorial_radius(cls):
+        # Semi-major axis of the WGS84 spheroid (meters)
+        return 6378137.0
+
+    @classmethod
+    def polar_radius(cls):
+        # Semi-minor axis of the wgs84 spheroid (meters)
+        return 6356752.314245
+
+    @classmethod
+    def first_eccentricity_parameter(cls):
+        # https://en.wikipedia.org/wiki/Eccentricity_(mathematics)#Ellipses
+        return math.sqrt(1.0 - (Ecef.polar_radius() ** 2) / (Ecef.equatorial_radius() ** 2))
+
+    @classmethod
+    def second_eccectricity_parameter(cls):
+        # https://en.wikipedia.org/wiki/Eccentricity_(mathematics)#Ellipses
+        return math.sqrt((Ecef.equatorial_radius() ** 2) / (Ecef.polar_radius() ** 2) - 1.0)
+
+    def ecef_to_latlon(self):
+        lon_in_radians = math.atan2(self.y, self.x)
+        lon_in_degrees = math.degrees(lon_in_radians)
+        xy_norm = math.sqrt(self.x ** 2 + self.y ** 2)
+        a = Ecef.equatorial_radius()
+        b = Ecef.polar_radius()
+        p = Ecef.second_eccectricity_parameter()
+        e = Ecef.first_eccentricity_parameter()
+        angle = math.atan((self.z * a) / (xy_norm * b))
+        lat_in_radians = math.atan((self.z + ((p ** 2) * b * (math.sin(angle) ** 3))) /
+                                   (xy_norm - (e ** 2) * a * (math.cos(angle) ** 3)))
+        lat_in_degrees = math.degrees(lat_in_radians)
+        return LatLon(lat_in_degrees, lon_in_degrees)
+
+    def __eq__(self, other):
```

Comment:
I did something similar with BoundingBox: hash(hash(origin), hash(corner))
For Enu I have just copied the point implementation: hash((hash(self.e) << 4) + hash(self.n) + (hash(self.u) ^ 0xFFFFFF))
Or should I do hash((self.e, self.n, self.u)) or hash (hash(self.e), hash(self.n), hash(self.u))?
My question was if I should add the hash to LatLon, because this class still hasn't that method.